server/server.go

```diff
@@ -1101,6 +1119,8 @@ func (s *Server) campaignLeader() {
 	ctx, cancel := context.WithCancel(s.serverLoopCtx)
 	defer cancel()
 	go lease.KeepAlive(ctx)
+	s.SetLease(lease)
```

Comment:
Does this duplicate with the follower code?
```
     s.tso.SyncTimestamp
     defer s.tso.ResetTimestamp()
```

Reply:
This is used to set lease which we will use for test. `defer s.tso.ResetTimestamp()` will reset the lease inside the tso which help to check leader when we allocate TSOs.

Reply:
got